Tests/SwiftCrossUITests/SwiftCrossUIBackendTests.swift

```diff
@@ -0,0 +1,158 @@
+            let updateDuration = 0.02
+            // Delay between observation-causing state mutations
+            let mutationGap = 0.01
+
+            let state = MyState()
+            let updateCount = Count()
+
+            let backend = await AppKitBackend()
+            let cancellable = state.didChange.observeAsUIUpdater(backend: backend) {
+                Task {
+                    await updateCount.update { $0 + 1 }
+                }
+                // Simulate an update of duration `updateDuration` seconds
+                Thread.sleep(forTimeInterval: updateDuration)
+            }
+            _ = cancellable  // Silence warning about cancellable being unused
+
+            let start = ProcessInfo.processInfo.systemUptime
+            for _ in 0..<mutationCount {
+                state.count += 1
+                try? await Task.sleep(for: .seconds(mutationGap))
+            }
+            let elapsed = ProcessInfo.processInfo.systemUptime - start
+
+            // Compute percentage of main thread's time taken up by updates.
+            let ratio = Double(await updateCount.count) * updateDuration / elapsed
+            #expect(
+                ratio <= 0.85,
+                """
+                Expected throttled updates to take under 85% of the main \
+                thread's time. Took \(Int(ratio * 100))%
+
+                Duration: \(elapsed), \(start) <=> \(end)
+                Updates: \(updateCount.count)/\(mutationCount)
+                """
+            )
+        }
+
+        @Test(
+            "A Basic Layout Works properly",
+            tags(.Layout)
+        )
+        @MainActor
+        func BasicLayout() async throws {
+            let backend = AppKitBackend()
+            let window = backend.createWindow(withDefaultSize: SIMD2(200, 200))
+
+            // Idea taken from https://github.com/pointfreeco/swift-snapshot-testing/pull/533
+            // and implemented in AppKitBackend.
+            window.backingScaleFactorOverride = 1
+            window.colorSpace = .genericRGB
+
+            let environment = EnvironmentValues(backend: backend)
+                .with(\.window, window)
+            let viewGraph = ViewGraph(
+                for: CounterView(),
+                backend: backend,
+                environment: environment
+            )
+            backend.setChild(ofWindow: window, to: viewGraph.rootNode.widget.into())
+
+            let result = viewGraph.update(
+                proposedSize: SIMD2(200, 200),
+                environment: environment,
+                dryRun: false
+            )
+            let view: AppKitBackend.Widget = viewGraph.rootNode.widget.into()
+            backend.setSize(of: view, to: result.size.size)
+            backend.setSize(ofWindow: window, to: result.size.size)
+
+            // MARK: This Might be a logic error as `result` was proposed a Size of `SIMD2(200, 200)`, the maximum allotted to it's parent window
```

Comment:
Child views decide their own size and may be smaller or larger than the proposed size given to them. This is expected 👍